### PR TITLE
Add cancellation to HtmlCopy GetAllClassificationSpans

### PR DESCRIPTION
**Pull request by KirillOsenkov**
Fixes https://github.com/microsoft/VS-PPT/issues/242
Fixes https://github.com/microsoft/VS-PPT/issues/239

### CODE REVIEW

**Comment by KirillOsenkov** [comment]
Hmm I fear if clicking Cancel does nothing then cancellation token might not be checked too often:
https://github.com/microsoft/VS-PPT/issues/242#issuecomment-791961049

Still I think it’s a good change to make but we need to keep thinking

**Comment by KirillOsenkov** [comment]
@metasyntactic @olegtk

File: src/CopyAsHtml/CopyAsHtml/FormattedStringBuilder.cs [hunk 1]
```diff
@@ -97,16 +98,24 @@ private void Append(string text)
...
         private IList<ClassificationSpan> GetClassificationSpansSync(SnapshotSpan parentSpan)
         {
             IAccurateClassifier accurateClassifier = _classifier as IAccurateClassifier;
 
             if (accurateClassifier != null && _waitIndicator != null)
             {
-                using (var waitContext = WaitHelper.Wait(_waitIndicator, "HTML Copy", "Formatting document for copying"))
+                try
+                {
+                    using (var waitContext = WaitHelper.Wait(_waitIndicator, "HTML Copy", "Formatting document for copying"))
```
Comment (olegtk) [inline src/CopyAsHtml/CopyAsHtml/FormattedStringBuilder.cs:109]: Do we know which component doesn't react to this cancellation token? I think I've seen a comment saying Cancel button doesn't work.

Comment (KirillOsenkov) [reply src/CopyAsHtml/CopyAsHtml/FormattedStringBuilder.cs:109]: Yes that's what worries me too. Eventually it's the taggers and the aggregate tagger and GetTags().

Another option is to drop AccurateTagger support entirely here. If a regular copy is working fine for them, this should too.

Comment (olegtk) [reply src/CopyAsHtml/CopyAsHtml/FormattedStringBuilder.cs:109]: I think @gundermanc has UI delay bug related to this, where LSP client makes remote call to get this information.
I'd suggest dropping IAccurateTagger altogether. The impact is that in this case you would get what you see, which should be good enough.

Comment (olegtk) [reply src/CopyAsHtml/CopyAsHtml/FormattedStringBuilder.cs:109]: Btw, we build PPT from an internal repo for legacy reasons

Comment (KirillOsenkov) [reply src/CopyAsHtml/CopyAsHtml/FormattedStringBuilder.cs]: I rewrote it to just delete IAccurateTagger

**Pull request closed without merging**